src/bundles/files.js

```diff
@@ -245,8 +245,9 @@ export default (opts = {}) => {
       }
     },
 
-    doFilesWrite: make(actions.WRITE, async (ipfs, root, rawFiles, id, { dispatch }) => {
-      const { streams, totalSize } = await filesToStreams(rawFiles)
+    doFilesWrite: make(actions.WRITE, async (ipfs, root, filesOrPromise, id, { dispatch }) => {
+      let files = filesOrPromise.then ? await filesOrPromise : filesOrPromise
```

Comment:
```suggestion
      let files = await filesOrPromise
```

If it's a promise it will wait for its resolution, if it's something else it will return immediately, so no need for this check.

Reply:
❤️  thanks @fsdiogo, that's neat ❤️